app/views/developers/index.html.erb

```diff
@@ -1,22 +1,35 @@
 <p id="notice"><%= notice %></p>
 
 <h1>Developers</h1>
+<!-- search  form-->
+  <%= form_for  @developer_search_form, url: developers_path, method: :get do |f| %>
```

Comment:
Can you add 100 developers and do the search?
And can you share what you see in the log (log/development.log) when you do the search?

Reply:
@whoait 
check  N + 1 queries
```
https://semaphoreci.com/blog/2017/08/09/faster-rails-eliminating-n-plus-one-queries.html
```

Reply:
@jp-ryuji @thienhv : I has fix it

Reply:
@whoait 
Please do what I asked.
> Can you add 100 developers and do the search?
> And can you share what you see in the log (log/development.log) when you do the search?

I see the following log.
And this is something I asked Hai shared with the other developers.
https://github.com/haiphantony/ws_exercise/pull/3#discussion_r178705126
```
Started GET "/" for 127.0.0.1 at 2018-04-06 16:26:43 +0900
   (0.3ms)  SET NAMES utf8,  @@SESSION.sql_mode = CONCAT(CONCAT(@@sql_mode, ',STRICT_ALL_TABLES'), ',NO_AUTO_VALUE_ON_ZERO'),  @@SESSION.sql_auto_is_null = 0, @@SESSION.wait_timeout = 2147483
   (0.2ms)  SELECT `schema_migrations`.`version` FROM `schema_migrations` ORDER BY `schema_migrations`.`version` ASC
Processing by DevelopersController#index as HTML
  Rendering developers/index.html.erb within layouts/application
   (0.2ms)  SELECT `languages`.`code`, `languages`.`id` FROM `languages`
   (0.2ms)  SELECT `programming_languages`.`name`, `programming_languages`.`id` FROM `programming_languages`
  Developer Load (0.4ms)  SELECT `developers`.* FROM `developers` ORDER BY `developers`.`created_at` DESC
   (0.3ms)  SELECT `languages`.`code` FROM `languages` INNER JOIN `developer_languages` ON `languages`.`id` = `developer_languages`.`language_id` WHERE `developer_languages`.`developer_id` = 51
   (0.4ms)  SELECT `programming_languages`.`name` FROM `programming_languages` INNER JOIN `developer_programming_languages` ON `programming_languages`.`id` = `developer_programming_languages`.`programming_language_id` WHERE `developer_programming_languages`.`developer_id` = 51
   (0.5ms)  SELECT `languages`.`code` FROM `languages` INNER JOIN `developer_languages` ON `languages`.`id` = `developer_languages`.`language_id` WHERE `developer_languages`.`developer_id` = 99
   (0.5ms)  SELECT `programming_languages`.`name` FROM `programming_languages` INNER JOIN `developer_programming_languages` ON `programming_languages`.`id` = `developer_programming_languages`.`programming_language_id` WHERE `developer_programming_languages`.`developer_id` = 99
   (0.2ms)  SELECT `languages`.`code` FROM `languages` INNER JOIN `developer_languages` ON `languages`.`id` = `developer_languages`.`language_id` WHERE `developer_languages`.`developer_id` = 98
   (0.4ms)  SELECT `programming_languages`.`name` FROM `programming_languages` INNER JOIN `developer_programming_languages` ON `programming_languages`.`id` = `developer_programming_languages`.`programming_language_id` WHERE `developer_programming_languages`.`developer_id` = 98
   (0.3ms)  SELECT `languages`.`code` FROM `languages` INNER JOIN `developer_languages` ON `languages`.`id` = `developer_languages`.`language_id` WHERE `developer_languages`.`developer_id` = 97
   (1.1ms)  SELECT `programming_languages`.`name` FROM `programming_languages` INNER JOIN `developer_programming_languages` ON `programming_languages`.`id` = `developer_programming_languages`.`programming_language_id` WHERE `developer_programming_languages`.`developer_id` = 97
   (0.3ms)  SELECT `languages`.`code` FROM `languages` INNER JOIN `developer_languages` ON `languages`.`id` = `developer_languages`.`language_id` WHERE `developer_languages`.`developer_id` = 96
   (0.4ms)  SELECT `programming_languages`.`name` FROM `programming_languages` INNER JOIN `developer_programming_languages` ON `programming_languages`.`id` = `developer_programming_languages`.`programming_language_id` WHERE `developer_programming_languages`.`developer_id` = 96
   (0.2ms)  SELECT `languages`.`code` FROM `languages` INNER JOIN `developer_languages` ON `languages`.`id` = `developer_languages`.`language_id` WHERE `developer_languages`.`developer_id` = 95
   (0.2ms)  SELECT `programming_languages`.`name` FROM `programming_languages` INNER JOIN `developer_programming_languages` ON `programming_languages`.`id` = `developer_programming_languages`.`programming_language_id` WHERE `developer_programming_languages`.`developer_id` = 95
   (0.2ms)  SELECT `languages`.`code` FROM `languages` INNER JOIN `developer_languages` ON `languages`.`id` = `developer_languages`.`language_id` WHERE `developer_languages`.`developer_id` = 94
   (0.5ms)  SELECT `programming_languages`.`name` FROM `programming_languages` INNER JOIN `developer_programming_languages` ON `programming_languages`.`id` = `developer_programming_languages`.`programming_language_id` WHERE `developer_programming_languages`.`developer_id` = 94
   (0.2ms)  SELECT `languages`.`code` FROM `languages` INNER JOIN `developer_languages` ON `languages`.`id` = `developer_languages`.`language_id` WHERE `developer_languages`.`developer_id` = 93
   (0.2ms)  SELECT `programming_languages`.`name` FROM `programming_languages` INNER JOIN `developer_programming_languages` ON `programming_languages`.`id` = `developer_programming_languages`.`programming_language_id` WHERE `developer_programming_languages`.`developer_id` = 93
   (0.2ms)  SELECT `languages`.`code` FROM `languages` INNER JOIN `developer_languages` ON `languages`.`id` = `developer_languages`.`language_id` WHERE `developer_languages`.`developer_id` = 92
   (0.3ms)  SELECT `programming_languages`.`name` FROM `programming_languages` INNER JOIN `developer_programming_languages` ON `programming_languages`.`id` = `developer_programming_languages`.`programming_language_id` WHERE `developer_programming_languages`.`developer_id` = 92
   (0.3ms)  SELECT `languages`.`code` FROM `languages` INNER JOIN `developer_languages` ON `languages`.`id` = `developer_languages`.`language_id` WHERE `developer_languages`.`developer_id` = 91
   (0.4ms)  SELECT `programming_languages`.`name` FROM `programming_languages` INNER JOIN `developer_programming_languages` ON `programming_languages`.`id` = `developer_programming_languages`.`programming_language_id` WHERE `developer_programming_languages`.`developer_id` = 91
```

Reply:
@jp-ryuji : Sorry I missed a case: search with all Developer. I has fix it. You can check.

Reply:
![14](https://user-images.githubusercontent.com/10244751/38409734-b62b33c0-39ac-11e8-82eb-c09d302af88a.PNG)